data_structures/src/chain.rs

```diff
@@ -2722,21 +2894,21 @@ impl TransactionsPool {
     /// pool.insert(transaction.clone(),0);
     /// pool.insert(transaction, 0);
     ///
-    /// let mut iter = pool.vt_iter();
+    /// let mut iter = pool.vt_and_sh_iter();
     /// let tx1 = iter.next();
     /// let tx2 = iter.next();
     ///
     /// ```
-    pub fn vt_iter(&self) -> impl Iterator<Item = &VTTransaction> {
-        self.sorted_vt_index
+    pub fn vt_and_sh_iter(&self) -> impl Iterator<Item = &Transaction> {
+        self.sorted_vt_and_sh_index
             .iter()
             .rev()
-            .filter_map(move |(_, h)| self.vt_transactions.get(h).map(|(_, t)| t))
+            .filter_map(move |(_, h)| self.vt_and_sh_transactions.get(h).map(|(_, t)| t))
     }
 
     /// An iterator visiting all the data request transactions
     /// in the pool
-    pub fn dr_iter(&self) -> impl Iterator<Item = &DRTransaction> {
+    pub fn dr_iter(&self) -> impl Iterator<Item = &Transaction> {
```

Comment:
Maybe it's better to leave this as a `DRTransaction` iterator:

```suggestion
    pub fn dr_iter(&self) -> impl Iterator<Item = &DRTransaction> {
```

This can be done using a simple `match t { DR(x) => x, _ => unreachable!() }` in the `.map` below

Reply:
Regarding transaction_fee function now needs a Transaction as an argument, it could be easier on this way